packages/lesswrong/server/migrations/20250411T194906.fix_incorrectly_nullable_fields.ts

```diff
@@ -0,0 +1,112 @@
+  await updateDefaultValue(db, ForumEvents, 'lightColor');
+
+  // eslint-disable-next-line no-console
+  console.log('Updating Posts.inactive default value');
+
+  // Posts.inactive - update default value
+  await updateDefaultValue(db, Posts, 'inactive');
+
+  // eslint-disable-next-line no-console
+  console.log('Revisions.inactive - drop removed field');
+
+  // Revisions.inactive - drop removed field
+  await dropRemovedField(db, Revisions, 'inactive');
+
+  // eslint-disable-next-line no-console
+  console.log('Tags.needsReview and descriptionTruncationCount - update null values to default and make not null');
+
+  // Tags.needsReview and descriptionTruncationCount - update null values to default and make not null
+  await db.none('UPDATE "Tags" SET "needsReview" = FALSE WHERE "needsReview" IS NULL');
+  await db.none('UPDATE "Tags" SET "descriptionTruncationCount" = 0 WHERE "descriptionTruncationCount" IS NULL');
+
+  await db.none(`ALTER TABLE "Tags"
+    ALTER COLUMN "needsReview" SET NOT NULL,
+    ALTER COLUMN "descriptionTruncationCount" SET NOT NULL
+  `);
+
+  // eslint-disable-next-line no-console
+  console.log('Tags.oldSlugs - set default value, update null values to default, and make not null');
+
+  // Tags.oldSlugs - set default value, update null values to default, and make not null
+  await db.none(`ALTER TABLE "Tags" ALTER COLUMN "oldSlugs" SET DEFAULT '{}'::TEXT[]`);
+  await db.none(`UPDATE "Tags" SET "oldSlugs" = '{}'::TEXT[] WHERE "oldSlugs" IS NULL`);
+  await db.none(`ALTER TABLE "Tags" ALTER COLUMN "oldSlugs" SET NOT NULL`);
+
+  // eslint-disable-next-line no-console
+  console.log('TagRels.inactive - update default value');
+
+  // TagRels.inactive - update default value
+  await updateDefaultValue(db, TagRels, 'inactive');
+
+  // eslint-disable-next-line no-console
+  console.log('Users.isAdmin - update default value');
+
+  // Users.isAdmin - update default value
+  await updateDefaultValue(db, Users, 'isAdmin');
+
+  // eslint-disable-next-line no-console
+  console.log('Users.slug, notificationKarmaPowersGained, and notificationNewDialogueChecks - make not null');
+
+  // Users.slug, notificationKarmaPowersGained, and notificationNewDialogueChecks - make not null
+  await db.none(`ALTER TABLE "Users"
+    ALTER COLUMN "slug" SET NOT NULL,
+    ALTER COLUMN "notificationKarmaPowersGained" SET NOT NULL,
+    ALTER COLUMN "notificationNewDialogueChecks" SET NOT NULL
+  `);
+
+  // eslint-disable-next-line no-console
+  console.log('Users.oldSlugs - set default value, update null values to default, and make not null');
+
+  // Users.oldSlugs - set default value, update null values to default, and make not null
+  await db.none(`ALTER TABLE "Users" ALTER COLUMN "oldSlugs" SET DEFAULT '{}'::TEXT[]`);
+  await db.none(`UPDATE "Users" SET "oldSlugs" = '{}'::TEXT[] WHERE "oldSlugs" IS NULL`);
+  await db.none(`ALTER TABLE "Users" ALTER COLUMN "oldSlugs" SET NOT NULL`);
```

Comment:
This part took a little over a minute in dev, in an operation that I'm pretty sure locked the Users table, so it might be worth doing it out-of-band (and not inside of a transaction) on prod before deploying, to reduce downtime.